addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -2144,7 +2144,14 @@ var SnippetsMenu = Widget.extend({
     _computeSelectorFunctions: function (selector, exclude, target, noCheck, isChildren, excludeParent) {
         var self = this;
 
-        exclude += `${exclude && ', '}.o_snippet_not_selectable`;
+        // TODO the `:not([contenteditable="true"])` part is designed to make
+        // images with such attribute editable even when they are in an
+        // environment where editing is not normally possible. This should be
+        // reviewed if we are to handle more hierarchy of editable nodes being
+        // editable despite their non editable environment.
+        // The `:not([data-snippet])` part is useful for snippets with the
+        // `o_not_editable` class (e.g. s_social_media).
+        exclude += `${exclude && ', '}.o_snippet_not_selectable, .o_not_editable:not([data-snippet]), .o_not_editable:not([data-snippet]) :not([contenteditable="true"])`;
```

Comment:
@qsm-odoo

- For the "first one", I agree with you that it couldn't be added because we don't need it to fix the issue that this PR exists for. And it could potentially create some changes that we haven't thought of. So [there is the PR](https://github.com/odoo/odoo/pull/117157) to revert it in the initial PR of this forward-port.

- About the "second one", the main goal of the `:not([data-snippet])` was to be able to edit icons in the social media snippet. But I replaced it by a more specific selector `:not(.s_social_media)` (waiting for a more proper fix).